src/components/ToolBar/AppMenu/menu-factory.tsx

```diff
@@ -427,27 +429,80 @@ export const AppMenuItemDialog: React.FC<AppMenuItemProps> = (props) => {
     </Box>
   )
 
-  return (
-    <Dialog
-      maxWidth="sm"
-      fullWidth
-      open={open}
-      onClose={handleClose}
-      onKeyDown={(e) => {
-        e.stopPropagation()
-      }}
-      onClick={(e) => {
-        e.stopPropagation()
-      }}
-    >
-      <Box sx={{ p: 3.5 }}>
-        <Typography variant="h5">{app.name}</Typography>
+return (
+  <Dialog
+    maxWidth="sm"
+    fullWidth
+    open={open}
+    onClose={handleClose}
+    onKeyDown={(e) => {
+      e.stopPropagation();
+    }}
+    onClick={(e) => {
+      e.stopPropagation();
+    }}
+  >
+    <Box sx={{ p: 3.5 }}>
+      <Stack spacing={1.5}>
+        <Typography variant="h5">
+          {app.name}
+        </Typography>
+
+        <Typography variant="body1">
+  {app.showDescriptionInDialog ? (app.description || "No description available.") : null}
+</Typography>
+
         {inputDefinition}
-        {parametersSection}
-        {submitButton}
-      </Box>
-    </Dialog>
-  )
+
+        {/* ↓↓↓ MODIFIED parametersSection rendering below ↓↓↓ */}
+        {app.parameters && (
+  <Box sx={{ p: 3 }}>
+    {/* Hide "parameters" header */}
+    {app.parameters.map((parameter: ServiceAppParameter) => {
+      const isSpecialCheckbox =
+        app.name?.toLowerCase() === "networkx analyzer";
+
+      return (
+        <Box key={parameter.displayName} sx={{ mb: 2 }}>
+          {isSpecialCheckbox ? (
+            // ✅ Use your app’s logic but change layout
+            
+              <Box sx={{ display: 'inline-flex', alignItems: 'center' }}>
+                <Box sx={{ whiteSpace: 'nowrap' }}>
+                  {renderParameter(parameter)}
+                </Box>
+              </Box>
+            
+          ) : (
+            renderParameter(parameter)
+          )}
+        </Box>
+      );
+    })}
+  </Box>
+)}
+        {/* ↑↑↑ END parametersSection override ↑↑↑ */}
+        <Box sx={{ display: 'flex', justifyContent: 'space-between', alignItems: 'flex-end', mt: 3 }}>
+  {app.name?.toLowerCase() === "networkx analyzer" ? (
```

Comment:
Need to remove this as well since it is a one off. We could put the link in the description.